src/components/AudioPlayer/ProgressBar.js

```diff
@@ -1,231 +1,212 @@
-  constructor(props) {
-    super(props)
-    this.state = {
-      seeking: false,
-      seekingValue: 0,
-      ending: false,
-    }
-  }
+const ProgressBar = props => {
+  const { position, duration } = useProgress()
+  const [seeking, setSeeking] = useState(false)
+  const [seekingValue, setSeekingValue] = useState(0)
+  const [ending, setEnding] = useState(false)
+  const [recentlySeeked, setRecentlySeeked] = useState(false)
+  const [endActionRan, setEndActionRan] = useState(false)
 
   // When the user clicks and holds on the slider
-  startSeek = () => {
-    const { progress } = this.props
-    this.setState({ seeking: true, seekingValue: progress })
+
+  const startSeek = () => {
+    setSeeking(true)
+    setSeekingValue(position)
   }
   // When the user changes the value but hasn't let go of the slider yet
-  seek = values => {
-    this.setState({ seekingValue: values[0] })
+  const seek = values => {
+    setSeekingValue(values[0])
   }
   // User has let go of the slider
-  endSeek = () => {
-    this.setState({
-      seeking: false,
-      progress: seekingValue,
-      // Recently seeked is for logic concerning the value of the slider.
-      // Used to address issue where the slider would randomly jump back to
-      // where it was before the user seeked for a brief moment in time.
-      recentlySeeked: true,
-    })
-    const { updateProgress, updatePlayed } = this.props
-    const { duration, seekingValue } = this.state
+  const endSeek = () => {
+    setSeeking(false)
+    // Recently seeked is for logic concerning the value of the slider.
+    // Used to address issue where the slider would randomly jump back to
+    // where it was before the user seeked for a brief moment in time.
+    setRecentlySeeked(true)
+    const { updateProgress, updatePlayed } = props
     let seekValueSec = seekingValue * duration
     updatePlayed(seekValueSec)
     updateProgress(seekingValue)
     TrackPlayer.seekTo(seekValueSec)
   }
 
   // resets progress to zero and calls on the action for the end of a song
-  endTrack = async () => {
-    const { updatePlayed, updateProgress, topScreen, endSong } = this.props
-    this.setState({ ending: true })
-
+  const endTrack = async () => {
+    const {
+      updatePlayed,
+      updateProgress,
+      topScreen,
+      endSong,
+      updatePlaying,
+    } = props
+    console.log('end tracK')
+    setEnding(true)
+    setEndActionRan(true)
+
+    updatePlaying(false)
     updateProgress(0)
     updatePlayed(0)
     await TrackPlayer.seekTo(0)
     if (topScreen) endSong()
 
-    this.setState({ ending: false })
+    setEnding(false)
   }
 
-  // In the mobile version, played, duration, and progress are all updated by
-  // the render function in ProgressBar, not the AudioPlayerSub, so if there are differences
-  // between the new props and the old props, don't rerender.
-  // Prevents an infinite loop.
-  shouldComponentUpdate(nextProps) {
```

Comment:
I'm having trouble seeing it, did you replicate this code to work in the functional component? If not, have you seen any issues during testing?

Reply:
I haven't come across any issues here yet, I think moving to a functional component made much of what was happening here unecessary